background/services/name/resolvers/uns.ts

```diff
@@ -0,0 +1,162 @@
+  const response = await fetch(
+    `https://unstoppabledomains.g.alchemy.com/domains/?owners=${address}&sortBy=id&sortDirection=DESC`,
+    {
+      method: "GET",
+      headers: {
+        Authorization: `Bearer ${process.env.ALCHEMY_KEY?.trim()}`,
+      },
+    }
+  )
+  const data = await response.json()
+
+  return data
+}
+
+/**
+ * Check if a given string a valid UNS domain
+ */
+const isValidUNSDomainName = (s: string): boolean => {
+  const trimmedString = s.trim()
+
+  // Any valid domain name will have a period as part of the string
+  if (trimmedString.lastIndexOf(".") > 0) {
+    const domainExtension = trimmedString.slice(trimmedString.lastIndexOf("."))
+    const supportedUNSDomains = [
+      ".crypto",
+      ".coin",
+      ".wallet",
+      ".bitcoin",
+      ".888",
+      ".nft",
+      ".dao",
+      ".zil",
+      ".x",
+    ]
+
+    if (supportedUNSDomains.includes(domainExtension)) {
+      return true
+    }
+  }
+  return false
+}
+
+export default function unsResolver(): NameResolver<"UNS"> {
+  return {
+    type: "UNS",
+    canAttemptNameResolution(): boolean {
+      return true
+    },
+    canAttemptAvatarResolution(): boolean {
+      return true
+    },
+    canAttemptAddressResolution({ name, network }: NameOnNetwork): boolean {
+      return sameNetwork(network, ETHEREUM) && isValidUNSDomainName(name)
+    },
+
+    async lookUpAddressForName({
+      name,
+      network,
+    }: NameOnNetwork): Promise<AddressOnNetwork | undefined> {
+      // We try to resolve the name using unstoppable domains resolution
+      const address = (await lookupUNSDomain(name))?.meta?.owner
+
+      if (address === undefined || address === null) {
+        return undefined
+      }
+
+      return {
+        address,
+        network,
+      }
+    },
+    async lookUpAvatar(
+      addressOrNameOnNetwork: AddressOnNetwork | NameOnNetwork
+    ) {
+      const { network } = addressOrNameOnNetwork
+      const { address } =
+        "address" in addressOrNameOnNetwork
+          ? addressOrNameOnNetwork
+          : (await this.lookUpAddressForName(addressOrNameOnNetwork)) ?? {
+              address: undefined,
+            }
+
+      if (address === undefined) {
+        return undefined
+      }
+
+      // Then we get all the records associated with the particular ETH address
+      const data = (await reverseLookupAddress(address))?.data
+
+      // check if any of the records returned has an NFT picture associated with it
+      let avatarUrn = data
+        .map(
+          (record: {
+            attributes: { records: { [x: string]: string | undefined } }
+          }) => record.attributes.records["social.picture.value"]
+        )
+        .find(isDefined)
+
+      if (avatarUrn === undefined || avatarUrn === null) {
+        return undefined
+      }
+      // modify the nft picture value and make it a correct avatar urn
+      avatarUrn = avatarUrn.replace(/^1\/(erc1155 || erc721)/, "eip155:1/$1")
```

Comment:
```suggestion
      avatarUrn = avatarUrn.replace(/^1\/(erc1155|erc721)/, "eip155:1/$1")
```

I believe this is the only way the regex will work :point_up_2: but haven't triple-checked.